docs/CodingConventions.md

```diff
@@ -0,0 +1,492 @@
+PREFACE
+=======
+
+This document describes a set of conventions that need to be honored during
+development of this project. Note that not necessarily the whole code
+currently follows all these conventions - but this is the case only of
+the older code that wasn't yet fixed. These conventions should be followed
+in any newly written code.
+
+These conventions are important for two reasons:
+
+1. If something can be written using various different variants as allowed
+by the language syntax, only one should be used overall in the code because
+otherwise readers get easily confused.
+
+2. Readability is most important when doing source code analysis for fixing
+bugs and finding the best way of adding a new feature. This is then essential
+for minimizing the time spent on that effort.
+
+There could be cases that sometimes it is required to break the convention, but
+if so, it should be an exception with a reasonable explanation that it supports
+the above principles better. If it doesn't contribute to the code this way, the
+convention shall be followed.
+
+
+The language standard
+=====================
+
+The C language, where it is applicable, is expected to be supported as C90
+standard (ISO/IEC 9899:1990, AKA ANSI C). This concerns both the subprojects
+(haicrypt) and examples written in C, as well as the C API of the SRT library.
+
+The SRT library sources (sources for libsrt) written in C++ should use the
+C++03 standard (ISO/IEC 14992:2003, which is C++98 with minor description fixes).
+This is required to support development projects with toolchains using very old
+compilers that do not support newer standards.
+
+The application support files (or library) as well as support and development
+applications, as well as C++ examples, should use C++11 standard (ISO/IEC
+14882:2011 - this may change to a higher standard in future).
+
+
+Technical syntax variants
+=========================
+
```

Comment:
Please use `#` for Heading 1, `##` for Heading 2, etc.
Those `===` and `----` produce non-conventional formatting.


![image](https://user-images.githubusercontent.com/12700120/70798743-13ac6d80-1da8-11ea-86e0-2e1b1a62f617.png)

Reply:
Right. Forgot that trick.